### PR TITLE
fix: allow `{{else}} {{#unless ...}}` in the 'simple-unless' rule and make corresponding update to 'no-negated-condition' rule.

### PR DESCRIPTION
**Pull request by bmish**
There are two relevant cases:
1. `{{#if condition}} ... {{else unless condition}} ... {{/if}}`
2. `{{#if condition}} ... {{else}} {{#unless condition}} ... {{/unless}} {{/if}}`

Currently, they are both banned by the 'simple-unless' rule, but I suspect that (2) was not originally intended to be banned (banning (2) may have been a mistake).

Here are the reasons why I don't think (2) was intended to be banned:
1. These two cases are difficult to distinguish with the AST.
2. The error message returned by both of these cases mentions (1) but not (2): 'Using an `{{else unless}}` block should be avoided.'
3. (1) is obviously undesirable whereas (2) isn't. In fact, I've heard multiple engineers voice surprise that (2) is currently banned.

As a result, I have updated 'simple-unless' to allow (2).

In addition, I have also update the related 'no-negated-condition' rule to allow suggesting (2), whereas previously it specifically avoided suggesting (2) to avoid conflicting with the 'simple-unless' rule.

### CODE REVIEW

File: test/unit/rules/lint-no-negated-condition.js [hunk 1]
```diff
@@ -181,6 +176,23 @@ generateRuleTests({
       },
     },
 
+    // Nested inside the body of an `else`:
+    {
+      template: `{{#if condition}}
+                 {{else}}
+                   {{! some comment }}
```
Comment (rwjblue) [inline test/unit/rules/lint-no-negated-condition.js]: Can you add another "bad" test without this comment (I'd like one with the comment and one without it) and without whitespace separating the curlies?

```hbs
{{#if condition}}{{else}}{{#if (not condition)}}<div></div>{{/if}}{{/if}}
```

In this circumstance the resulting AST is _nearly identical_ to:

```hbs
{{#if condition}}{{else if (not condition)}}<div></div>{{/if}}{{/if}}
```

And I want to make sure we are still allowing the latter (since it seems like that was your intent) while disallowing the former...

Comment (bmish) [reply test/unit/rules/lint-no-negated-condition.js]: Updated tests as you suggested.
